packages/mask/src/plugins/Referral/helpers/index.tsx

```diff
@@ -0,0 +1,62 @@
+import { padStart } from 'lodash-unified'
+import { BigNumber } from '@ethersproject/bignumber'
+import type { ChainId } from '@masknet/web3-shared-evm'
+import { createTypedMessageMetadataReader } from '@masknet/typed-message'
+
+import { META_KEY, supportedChainId } from '../constants'
+import type { ReferralMetaData, ChainAddress, ChainAddressProps, EvmAddress, Bytes32, Bytes24 } from '../types'
+import schema from '../schema.json'
+import { bi, Buffer20, Buffer24 } from './buffer'
+
+function toChainAddress(chainId: BigNumber | bigint, address: Buffer20): Buffer24 {
+    if (address.byteLength !== 20) throw new Error('invalid address')
+    const b = Buffer.alloc(24)
+    // Only numeric network id's are supported in the chain address, with max of uint32.
+    // TODO throw on this if wrong
+    b.writeUInt32BE(Number(bi(chainId)), 0)
```

Comment:
I want to say I prefer `DataView` but I don't know about if they support the BE part, so this depends on you.

Reply:
We use it in the other services. But, I will keep your suggestion about DataView in mind.